### DISCUSSION TITLE
Where can I find old versions of Z3?

### DISCUSSION THREAD

**Discussion by nhamlv-55** [Z3Prover/z3#7093] [Q&A]
Hello,
I am interested in trying Z3 with partitioning (as described in this paper https://link.springer.com/content/pdf/10.1007/978-3-642-02658-4_60.pdf )
However, this implementation seems no longer supported since Z3 4.0. Github has no version older than that, so could someone point me to where I can get them?
Also, if I remember correctly, old versions of Z3 are Windows binaries. Do the Linux versions exist anywhere?
Many thanks

**Comment by wintersteiger** [comment]
Some older Linux binaries are available on the SMT-COMP webpages and on [StarExec](https://www.starexec.org/), but I haven't found one that has the portfolio built in yet.

**Reply by nhamlv-55** [reply]
Thanks @wintersteiger ! Do you have old Windows binaries then?

**Reply by wintersteiger** [reply]
Sadly I don't! However, there is a very, very small chance that there is one in the [Codeplex archive](https://archive.org/details/sylirana_ms_codeplex_zips). I can't remember when exactly Z3 moved to Codeplex, but it's possible that it still contained the portfolio code at the time.
